apps/namadillo/src/atoms/registry/services.ts

```diff
@@ -0,0 +1,37 @@
+import { Asset, AssetList } from "@chain-registry/types";
+import { Coin } from "@cosmjs/launchpad";
+import { StargateClient } from "@cosmjs/stargate";
+import BigNumber from "bignumber.js";
+
+export type AssetWithBalance = {
+  asset: Asset;
+  balance?: BigNumber;
+};
+
+export const queryAssetBalances = async (
+  owner: string,
+  rpc: string
+): Promise<Coin[]> => {
+  const client = await StargateClient.connect(rpc);
+  const balances = (await client.getAllBalances(owner)) || [];
+  return balances as Coin[];
+};
+
+export const mapCoinsToAssets = (
+  coins: Coin[],
+  assetList: AssetList
+): Record<string, AssetWithBalance> => {
+  return coins.reduce((prev, current) => {
+    const asset = assetList.assets.find(
+      (asset) => asset.base === current.denom
+    );
+    if (!asset) return prev;
```

Comment:
I think we'll still want to display an asset even if we don't have information for it in the asset list. For example, NAM isn't going to show up in the asset list (although actually maybe we could add it), but we still want to support transferring NAM over IBC.

The easiest thing to do here might be to change TransferModule to specify the properties it needs for assets e.g. instead of `Asset`, use `Pick<Asset, "name" | "base">` or something like that. Then we can more easily construct what the TransferModule needs in these cases.

Reply:
I think this might be a bigger problem, because if the asset is not known in the asset list, maybe we will need to query some of its info from somewhere. Can we separate this into a new issue?